dnf5/commands/download/download.cpp

```diff
@@ -161,10 +182,25 @@ void DownloadCommand::run() {
     }
 
     if (url_option->get_value()) {
+        // If no urlprotocols are specified, then any urlprotocol is acceptable
+        if (urlprotocol_option.empty()) {
+            urlprotocol_option = urlprotocol_valid_options;
+        }
         for (auto & [nerva, pkg] : download_pkgs) {
             auto urls = pkg.get_remote_locations();
             libdnf_assert(!urls.empty(), "Failed to get mirror for package: \"{}\"", pkg.get_name());
-            std::cout << urls[0] << std::endl;
+            auto valid_url = std::find_if(urls.begin(), urls.end(), [this](std::string url) {
+                for (auto protocol : urlprotocol_option) {
+                    if (url.starts_with(protocol)) {
```

Comment:
You need to match up to and including a colon charater. Otherwise, "httpnextgen://foo" would match "http".

Reply:
It would be better if the filtering logic wasn't duplicated here, you can pass the set of protocols directly to `get_remote_locations()`, it takes it as an optional parameter.

It does have the same problem with the `httpnextgen://foo` but you could fix it there directly.

Reply:
Added to the get_remote_locations function